### DISCUSSION TITLE
Deployment Issue

### DISCUSSION THREAD

**Discussion by IvernBurner** [glpi-project/glpi-agent#656] [Q&A]
### Documentation & bug reporting acknowledgment

Yes, I read it

### Describe your problem

I'm having some troubles trying to deploy the agent on several PC using a GPO, this is a .vbs file and when I just execute it I'm having the following:

```
C:\TEMP>cscript GLPI-Agent_Deploy.vbs
Microsoft (R) Windows Script Host versión 5.812
Copyright (C) Microsoft Corporation. Reservados todos los derechos.

System architecture detected: x64
Setup architecture detected: x64
Installation needed: 1.7.3
Downloading: https://github.com/glpi-project/glpi-agent/releases/download/1.7.3/GLPI-Agent-1.7.3-x64.msi
Next attempt in 30 seconds...
Running: MsiExec.exe /i "C:\Users\mmontana\AppData\Local\Temp\GLPI-Agent-1.7.3-x64.msi" /quiet RUNNOW=1 SERVER='http://glpi.hormipresa.com'
Deployment failed! (Err=1603)
Scheduling: DEL /Q /F "C:\Users\mmontana\AppData\Local\Temp\GLPI-Agent-1.7.3-x64.msi"
```

**Comment by g-bougard** [comment]
`Next attempt in 30 seconds...` seems telling there's a network issue.

Are you sure internet connectivity is working from this computer ? If not, you'll probably have to download the package to a shared place and tell the script to use this shared place to download the package.

**Reply by IvernBurner** [reply]
I don't think there is a network issue, I can perfectly download it manually from the same device.

**Reply by IvernBurner** [reply]
As you can see, with the package in a shared place, It still giving me the same issue.

![image](https://github.com/glpi-project/glpi-agent/assets/74453883/48503379-e691-430e-83df-ba816475051c)

![image](https://github.com/glpi-project/glpi-agent/assets/74453883/e1963cda-795f-425b-aeb1-acca5ac5d9dd)

**Reply by g-bougard** [reply]
Sorry, but this is not the same issue: `Err=1603` & `Err=1619`

In the first case, check this official page: https://learn.microsoft.com/en-us/troubleshoot/windows-server/application-management/msi-installation-error-1603

According to this page: https://learn.microsoft.com/en-us/windows/win32/msi/error-codes
The 1619 error seems to tell us the installer can't open the package. And indeed, it's normal as `SetupLocation` must be a base url or a folder, not the full path to the package. I guess the vbs is attempting to open `C:\Temp\GLPI-Agent-1.7.3-x64.msi\GLPI-Agent-1.7.3-x64.msi` ... which of course doesn't exist.

**Reply by IvernBurner** [reply]
Right, now it is working but when I put it on scripts for the GPO, It is just doing nothing.

![image](https://github.com/glpi-project/glpi-agent/assets/74453883/79aa89f7-4e5b-4878-8c77-66c2bb5e04d9)

**Reply by IvernBurner** [reply]
Now I have discovered that using a vbs with the .msi executable, It needs admin rights.

**Reply by g-bougard** [reply]
Yes, you need administrator rights to install GLPI-Agent MSI package.

**Reply by IvernBurner** [reply]
I have resolved this issue, now I use the https and all my machines have lost the link with GLPI, that is because the SSL Check, I have discovered how to reconfigure the vbs using "NO_SSL_CHECK=1" but I need to reconfigure on all, I tried the following but is not working, any idea?

![image](https://github.com/glpi-project/glpi-agent/assets/74453883/9a55da61-890c-4e80-8612-309033a5a350)

**Reply by g-bougard** [reply]
Hi @IvernBurner 

you should not use yourself `REINSTALL=feat_AGENT` in `SetupOptions` but set `Reconfigure = "Yes"` just below and leave the script does its job.

In the VBS script logic, if you not set Reconfigure, it can decide to not run the installer: see https://github.com/glpi-project/glpi-agent/blob/1ea424d0a1112071fa8b6a944761ef04451f1c25/contrib/windows/glpi-agent-deployment.vbs#L588

**Reply by g-bougard** [reply]
Regarding the ssl question, can you check the "_How can I configure my agent to access my GLPI server using SSL?_" at the end of this page: https://faq.teclib.com/02_FAQ/Agent/#glpi-agent ?
